src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -47,12 +47,21 @@ object WorkflowManagerActor {
   final case class WorkflowMetadata(id: WorkflowId) extends WorkflowManagerActorMessage
   final case class RestartWorkflows(workflows: Seq[WorkflowDescriptor]) extends WorkflowManagerActorMessage
   final case class CallCaching(id: WorkflowId, parameters: QueryParameters, call: Option[String]) extends WorkflowManagerActorMessage
+  case object AbortAllWorkflows extends WorkflowManagerActorMessage
 
   def props(backend: Backend): Props = Props(new WorkflowManagerActor(backend))
 
   // How long to delay between restarting each workflow that needs to be restarted.  Attempting to
   // restart 500 workflows at exactly the same time crushes the database connection pool.
   lazy val RestartDelay = 200 milliseconds
+
+  sealed trait WorkflowManagerState
+  case object Running extends WorkflowManagerState
+  case object Aborting extends WorkflowManagerState
+  case object Done extends WorkflowManagerState
+
+  sealed trait WorkflowManagerData
+  case object NoData extends WorkflowManagerData
```

Comment:
The state data should include the workflow store, at which time the write-once class can be deleted.